File: Models/Notification.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace MilaAPI.Models
{
	public class Notification
	{
		[Key]
		public int Id { get; set; }
		
		[Required]
		public int UserId { get; set; }
		public User User { get; set; }

		public string Message { get; set; }  // Notification message content
		public DateTime DateCreated { get; set; }  // When the notification was created
		public bool IsRead { get; set; }  // Whether the user has read the notification
		public string Type { get; set; }  // Type of notification (e.g., "ExpenseAlert", "GoalReminder")
	}
}

File: front-end/milaapp/lib/models/notification.dart
class Notification {
  final String id;
  final String message;
  final DateTime dateCreated;
  final bool isRead;
  final String userId;

  Notification({
    required this.id,
    required this.message,
    required this.dateCreated,
    required this.isRead,
    required this.userId,
  });

  factory Notification.fromJson(Map<String, dynamic> json) {
    return Notification(
      id: json['id'],
      message: json['message'],
      dateCreated: DateTime.parse(json['dateCreated']),
      isRead: json['isRead'],
      userId: json['userId'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'message': message,
      'dateCreated': dateCreated.toIso8601String(),
      'isRead': isRead,
      'userId': userId,
    };
  }
}
